OptimizelySDK/Utils/ExtensionMethods.cs

```diff
@@ -0,0 +1,43 @@
+﻿/* 
+ * Copyright 2018, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Text;
+
+namespace OptimizelySDK.Utils
+{
+    public static class ExtensionMethods
```

Comment:
This seems like a very generic name. Can we call it `ExceptionExtensionMethods`?

This is actually a pretty cool feature of the platform/language btw.

Reply:
@mikeng13 I believe the name should be ExceptionDetailExtension, since it's class, looks odd to call with Methods? make sense?

Reply:
Renamed the class as ExceptionExtensions to contain any exception related extensions in it. Thoughts?

Reply:
Also making the class internal restricts us from writing its unit tests.